File: android/app/src/main/java/com/expensetracker/ui/components/FormComponents.kt
package com.expensetracker.ui.components

import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.KeyboardArrowRight
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import com.expensetracker.ui.theme.*

@Composable
fun ListRow(
    leadingIcon: ImageVector,
    title: String,
    subtitle: String? = null,
    rightValue: String? = null,
    showChevron: Boolean = true,
    onClick: (() -> Unit)? = null,
    modifier: Modifier = Modifier
) {
    PastelCard(
        modifier = modifier.then(
            if (onClick != null) Modifier.clickable { onClick() } else Modifier
        )
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.spacedBy(12.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            // Leading icon
            Icon(
                imageVector = leadingIcon,
                contentDescription = null,
                tint = PrimaryBlue,
                modifier = Modifier.size(24.dp)
            )
            
            // Title and subtitle
            Column(
                modifier = Modifier.weight(1f),
                verticalArrangement = Arrangement.spacedBy(2.dp)
            ) {
                Text(
                    text = title,
                    style = MaterialTheme.typography.bodyLarge,
                    color = MaterialTheme.colorScheme.onSurface
                )
                
                if (subtitle != null) {
                    Text(
                        text = subtitle,
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f)
                    )
                }
            }
            
            // Right value
            if (rightValue != null) {
                Text(
                    text = rightValue,
                    style = MaterialTheme.typography.labelLarge,
                    color = MaterialTheme.colorScheme.onSurface
                )
            }
            
            // Chevron
            if (showChevron) {
                Icon(
                    imageVector = Icons.Default.KeyboardArrowRight,
                    contentDescription = null,
                    tint = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.5f),
                    modifier = Modifier.size(16.dp)
                )
            }
        }
    }
}

@Composable
fun FormTextField(
    label: String,
    value: String,
    onValueChange: (String) -> Unit,
    placeholder: String,
    modifier: Modifier = Modifier,
    keyboardOptions: KeyboardOptions = KeyboardOptions.Default
) {
    Column(
        modifier = modifier,
        verticalArrangement = Arrangement.spacedBy(4.dp)
    ) {
        Text(
            text = label,
            style = MaterialTheme.typography.labelLarge,
            color = MaterialTheme.colorScheme.onSurface
        )
        
        OutlinedTextField(
            value = value,
            onValueChange = onValueChange,
            placeholder = { 
                Text(
                    placeholder,
                    style = MaterialTheme.typography.bodyLarge
                ) 
            },
            modifier = Modifier.fillMaxWidth(),
            keyboardOptions = keyboardOptions,
            colors = OutlinedTextFieldDefaults.colors(
                focusedBorderColor = PrimaryBlue,
                unfocusedBorderColor = CardBorder,
                focusedContainerColor = CardBackground,
                unfocusedContainerColor = CardBackground
            )
        )
    }
}

@Composable
fun MoneyTextField(
    label: String,
    value: String,
    onValueChange: (String) -> Unit,
    modifier: Modifier = Modifier
) {
    Column(
        modifier = modifier,
        verticalArrangement = Arrangement.spacedBy(4.dp)
    ) {
        Text(
            text = label,
            style = MaterialTheme.typography.labelLarge,
            color = MaterialTheme.colorScheme.onSurface
        )
        
        OutlinedTextField(
            value = value,
            onValueChange = onValueChange,
            placeholder = { 
                Text(
                    "0.00",
                    style = MaterialTheme.typography.bodyLarge
                ) 
            },
            leadingIcon = {
                Text(
                    "$",
                    style = MaterialTheme.typography.bodyLarge,
                    color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.7f)
                )
            },
            modifier = Modifier.fillMaxWidth(),
            keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Decimal),
            colors = OutlinedTextFieldDefaults.colors(
                focusedBorderColor = PrimaryBlue,
                unfocusedBorderColor = CardBorder,
                focusedContainerColor = CardBackground,
                unfocusedContainerColor = CardBackground
            )
        )
    }
}

@Composable
fun ToggleRow(
    title: String,
    subtitle: String? = null,
    checked: Boolean,
    onCheckedChange: (Boolean) -> Unit,
    modifier: Modifier = Modifier
) {
    PastelCard(modifier = modifier) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Column(
                modifier = Modifier.weight(1f),
                verticalArrangement = Arrangement.spacedBy(2.dp)
            ) {
                Text(
                    text = title,
                    style = MaterialTheme.typography.bodyLarge,
                    color = MaterialTheme.colorScheme.onSurface
                )
                
                if (subtitle != null) {
                    Text(
                        text = subtitle,
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f)
                    )
                }
            }
            
            Switch(
                checked = checked,
                onCheckedChange = onCheckedChange,
                colors = SwitchDefaults.colors(
                    checkedThumbColor = OnPrimary,
                    checkedTrackColor = PrimaryBlue,
                    uncheckedThumbColor = MaterialTheme.colorScheme.onSurface.copy(alpha = 0.6f),
                    uncheckedTrackColor = CardBorder
                )
            )
        }
    }
}

@Preview(showBackground = true)
@Composable
fun FormComponentsPreview() {
    ExpenseTrackerTheme {
        Column(
            modifier = Modifier.padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            ListRow(
                leadingIcon = Icons.Default.KeyboardArrowRight, // Using available icon for preview
                title = "Groceries",
                subtitle = "Today, 2:30 PM",
                rightValue = "-$45.67",
                onClick = {}
            )
            
            FormTextField(
                label = "Transaction Name",
                value = "",
                onValueChange = {},
                placeholder = "Enter name"
            )
            
            MoneyTextField(
                label = "Amount",
                value = "25.50",
                onValueChange = {}
            )
            
            ToggleRow(
                title = "Include Recurring Expenses",
                subtitle = "Add recurring expenses to budget calculations",
                checked = true,
                onCheckedChange = {}
            )
        }
    }
}

File: ios/ExpenseTracker/Components/FormComponents.swift
import SwiftUI

// MARK: - List Row

struct ListRow: View {
    let leadingIcon: String
    let title: String
    let subtitle: String?
    let rightValue: String?
    let showChevron: Bool
    let action: (() -> Void)?
    
    init(
        leadingIcon: String,
        title: String,
        subtitle: String? = nil,
        rightValue: String? = nil,
        showChevron: Bool = true,
        action: (() -> Void)? = nil
    ) {
        self.leadingIcon = leadingIcon
        self.title = title
        self.subtitle = subtitle
        self.rightValue = rightValue
        self.showChevron = showChevron
        self.action = action
    }
    
    var body: some View {
        let content = HStack(spacing: Theme.Spacing.md) {
            // Leading icon
            Image(systemName: leadingIcon)
                .font(.title3)
                .foregroundColor(Theme.Colors.primary)
                .frame(width: 24, height: 24)
            
            // Title and subtitle
            VStack(alignment: .leading, spacing: 2) {
                Text(title)
                    .font(Theme.Typography.body)
                    .foregroundColor(Theme.Colors.onSurface)
                
                if let subtitle = subtitle {
                    Text(subtitle)
                        .font(Theme.Typography.caption)
                        .foregroundColor(Theme.Colors.onSurface.opacity(0.6))
                }
            }
            
            Spacer()
            
            // Right value
            if let rightValue = rightValue {
                Text(rightValue)
                    .font(Theme.Typography.label)
                    .foregroundColor(Theme.Colors.onSurface)
            }
            
            // Chevron
            if showChevron {
                Image(systemName: "chevron.right")
                    .font(.caption)
                    .foregroundColor(Theme.Colors.onSurface.opacity(0.5))
            }
        }
        .padding(Theme.Spacing.lg)
        .background(
            RoundedRectangle(cornerRadius: Theme.Shapes.cardCornerRadius)
                .fill(Theme.Colors.card)
                .overlay(
                    RoundedRectangle(cornerRadius: Theme.Shapes.cardCornerRadius)
                        .stroke(Theme.Colors.cardBorder, lineWidth: 1)
                )
        )
        
        if let action = action {
            Button(action: action) {
                content
            }
            .buttonStyle(PlainButtonStyle())
        } else {
            content
        }
    }
}

// MARK: - Form Fields

struct FormTextField: View {
    let label: String
    @Binding var text: String
    let placeholder: String
    
    var body: some View {
        VStack(alignment: .leading, spacing: Theme.Spacing.xs) {
            Text(label)
                .font(Theme.Typography.label)
                .foregroundColor(Theme.Colors.onSurface)
            
            TextField(placeholder, text: $text)
                .font(Theme.Typography.body)
                .padding(Theme.Spacing.md)
                .background(
                    RoundedRectangle(cornerRadius: Theme.Shapes.buttonCornerRadius)
                        .fill(Theme.Colors.card)
                        .overlay(
                            RoundedRectangle(cornerRadius: Theme.Shapes.buttonCornerRadius)
                                .stroke(Theme.Colors.cardBorder, lineWidth: 1)
                        )
                )
        }
    }
}

struct MoneyField: View {
    let label: String
    @Binding var amount: String
    
    var body: some View {
        VStack(alignment: .leading, spacing: Theme.Spacing.xs) {
            Text(label)
                .font(Theme.Typography.label)
                .foregroundColor(Theme.Colors.onSurface)
            
            HStack {
                Text("$")
                    .font(Theme.Typography.body)
                    .foregroundColor(Theme.Colors.onSurface.opacity(0.7))
                
                TextField("0.00", text: $amount)
                    .font(Theme.Typography.body)
                    .keyboardType(.decimalPad)
            }
            .padding(Theme.Spacing.md)
            .background(
                RoundedRectangle(cornerRadius: Theme.Shapes.buttonCornerRadius)
                    .fill(Theme.Colors.card)
                    .overlay(
                        RoundedRectangle(cornerRadius: Theme.Shapes.buttonCornerRadius)
                            .stroke(Theme.Colors.cardBorder, lineWidth: 1)
                    )
            )
        }
    }
}

struct ToggleRow: View {
    let title: String
    let subtitle: String?
    @Binding var isOn: Bool
    
    init(title: String, subtitle: String? = nil, isOn: Binding<Bool>) {
        self.title = title
        self.subtitle = subtitle
        self._isOn = isOn
    }
    
    var body: some View {
        HStack {
            VStack(alignment: .leading, spacing: 2) {
                Text(title)
                    .font(Theme.Typography.body)
                    .foregroundColor(Theme.Colors.onSurface)
                
                if let subtitle = subtitle {
                    Text(subtitle)
                        .font(Theme.Typography.caption)
                        .foregroundColor(Theme.Colors.onSurface.opacity(0.6))
                }
            }
            
            Spacer()
            
            Toggle("", isOn: $isOn)
                .toggleStyle(SwitchToggleStyle(tint: Theme.Colors.primary))
        }
        .padding(Theme.Spacing.lg)
        .background(
            RoundedRectangle(cornerRadius: Theme.Shapes.cardCornerRadius)
                .fill(Theme.Colors.card)
                .overlay(
                    RoundedRectangle(cornerRadius: Theme.Shapes.cardCornerRadius)
                        .stroke(Theme.Colors.cardBorder, lineWidth: 1)
                )
        )
    }
}

#Preview {
    VStack(spacing: 16) {
        ListRow(
            leadingIcon: "creditcard",
            title: "Groceries",
            subtitle: "Today, 2:30 PM",
            rightValue: "-$45.67"
        ) {}
        
        ListRow(
            leadingIcon: "house",
            title: "Family Members",
            showChevron: true
        ) {}
        
        FormTextField(
            label: "Transaction Name",
            text: .constant(""),
            placeholder: "Enter name"
        )
        
        MoneyField(
            label: "Amount",
            amount: .constant("25.50")
        )
        
        ToggleRow(
            title: "Include Recurring Expenses",
            subtitle: "Add recurring expenses to budget calculations",
            isOn: .constant(true)
        )
    }
    .padding()
    .background(Theme.Colors.surface)
}